### PR TITLE
Create service for runtime metrics

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
This PR creates a gRPC service for exposing runtime metrics for workflows / nodes.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3272

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/event/event.proto [hunk 1]
```diff
@@ -104,6 +104,12 @@ message NodeExecutionEvent {
     // String location uniquely identifying where the deck HTML file is
     // NativeUrl specifies the url in the format of the configured storage provider (e.g. s3://my-bucket/randomstring/suffix.tar)
     string deck_uri = 19;
+
+    // This timestamp represents the instant when the event was reported by the executing framework. For example,
+    // when first processing a node the `occurred_at` timestamp should be the instant propeller makes progress, so when
+    // literal inputs are intially copied. The event however will not be sent until after the copy completes.
```
Comment (eapolinario) [inline protos/flyteidl/event/event.proto:110]: typo: intially

Comment (hamersaw) [reply protos/flyteidl/event/event.proto]: thanks! fixed.

File: protos/flyteidl/admin/common.proto [hunk 2]
```diff
@@ -300,3 +301,56 @@ message RawOutputDataConfig {
...
+// CategoricalSpanInfo represents a labelled Span, providing a brief attribution for the duration.
+message CategoricalSpanInfo {
+    // Category defines a collection of span categories.
+    enum Category {
+        UNKNOWN = 0;
+        NODE_TRANSITION = 1;
+        EXECUTION_OVERHEAD = 2;
+        EXECUTION_IDLE = 3;
+        PLUGIN_OVERHEAD = 4;
+        PLUGIN_RUNTIME = 5;
```
Comment (pingsutw) [inline protos/flyteidl/admin/common.proto:334]: qq: What's different between `PLUGIN_OVERHEAD` and `PLUGIN_RUNTIME`. 
Does `PLUGIN_OVERHEAD` mean the overhead in the propeller when creating the CR?
PLUGIN_RUNTIME is the actual task overhead?

**Review by pingsutw — APPROVED** [review]

File: protos/flyteidl/admin/common.proto (hunk 2 above)
Comment (hamersaw) [reply protos/flyteidl/admin/common.proto:334]: great question. so for example starting a k8s pod. `PLUGIN_OVERHEAD` is the time from when propeller performs the `Create` call to when the internal container starts. this covers things like pod scheduling, container pulls, etc. whereas `PLUGIN_RUNTIME`, in this example, is when the  container is actually running.

**Review by eapolinario — APPROVED** [review]

**Pull request merged**